### DISCUSSION TITLE
auto-scroll when dragging a rectangle selection in zoom level 50% and bottom right corner is outside of PD window?

### DISCUSSION THREAD

**Discussion by pstein** [tannerhelland/PhotoDemon#526] [General]
Assume I load a large image into PhotoDemon.
Since I want to see details I zoom into the image. So the current zoom level is e.g. 50%
Now I want to create a rectangle selecttion.
Since I zoomed in I can position the mouse exactly at the top left corner of the selection rectangle.
Now I drag the mouse (left mouse button is pressed) towards the bottom to image to release the mouse button 
when I reach the bottom right corner of the selection rectangle. The intended bottom right corner is currently not shown. 
It is outside the PD window.

I expected that PhotoDemon AUTOMATICALLY scrolls the image to the top when the mouse is reaching the lower window border.
BUT unfortunately this is not the case: NO AUTO_SCROLLING.

Very bad

I don't want to go to zoom level 100% since I cannot position the mouse exactly.

How can I enable auto-scrolling in this scenario?

PS: auto-scrolling speed should depend on how much the mouse is dragged outside the PD window.
When I position the mouse let say 3 pixels outside the PD window the scroll speed should be slow.

When I position the mouse 30 pixels outside the PD window the scroll speed should be fast

**Comment by hi5** [comment]
Autoscroll would be nice, but just in case:
* Use ctrl+mousewheel to zoom out - grab the selection corner, move it to "just about in the area" zoom in again, finalize selection
* You can use the cursor keys to scroll the image while continuing to select, it is of course pretty slow if you need to pan a large distance. 
* or just use the scrollbars or overview to move faster, you can can grab the corner again and continue your selection when you've moved it. 

A bit of kludge of course, but the ctrl+mousewheel works pretty decent
